book/website/reproducible-research/rdm/rdm-spreadsheets.md

```diff
@@ -1,35 +1,178 @@
 data for further analysis.
 
-It's often not practical to avoid the use of spreadsheets altogether but there are some simple steps that can be taken
-to mitigate their flaws.
-The following principles, provide some practical advice to ensure your data is clearly organised and human- and machine-readable {% cite Broman2018data --suppress_author %}:
-
-- Be consistent
-- Write dates as YYYY-MM-DD
-- Don't leave any cells empty, if there is no data use "NA" (Not Available)
-- Put each observation/sample in its own cell/row
-- Put each variable in a column
-- Each cell should contain information on one thing: separate multiple pieces of information to different cells
-- Create a data dictionary
-- Don't include calculations in the raw data files: make the file with raw data read-only!
-- Don’t use font, color or highlighting, consider adding another variable/column for this information
-- Choose descriptinve names for observations and variables and make sure there are no spaces or special characters in the names
-- Make backups
-- Use data validation to avoid data entry mistakes
-- Save the data in text files (such as `.csv` - comma-separated values) to ensure interoperability with other software programmes.
-
-You can explain the file naming convention in a README.txt file, so that it will also become clear to others what the file names mean.
-To learn more about data organisation in spreadsheets, have a look at the Data Carpentry lessons for [Social Scientists](https://datacarpentry.org/spreadsheets-socialsci/) and [Ecologists](https://datacarpentry.org/spreadsheet-ecology-lesson/).
+Indeed, a large part of the work of data scientist is to transform the data into a form that the computer can read. 
+This is particularly time consuming when the information is split between several spreadsheets and that no plan were made before data acquisition.
+
+There are very simple rules to facilitate data use, which go into the concept of [**tidy data**](https://en.wikipedia.org/w/index.php?title=Tidy_data&oldid=962241815). 
+Furthermore, this specific format also allows for filtering and sorting data easily in spreadsheet software. 
+In short:
+
+- one colum = one variable (no more, no less, this implies that two header names can not be identical)
+- one row = one sample
+- one cell = one information
+- **first row and only the first row is the header** 
+- Headers name: do not include special character (including space), do not start with a number
+
+| ![tidy table figure](../../figures/tidy-1.png)         |
+| ------------------------------------------------------------------------------------ |
+|Three rules which make a dataset tidy:|
+|1. Each variable must have its own column.|
+|2. Each observation must have its own row.|
+|3. Each value must have its own cell.  |
+
+There are data validation tools available, like https://goodtables.io, that allows you to check automatically whether your spreadsheets are tidy.
+
```

Comment:
```suggestion
## Consistent values

Especially if you work with several spreadsheets or if you work with a team during data collection, it is very important to make sure the same information will be entered with the same term, and a same term always convey the same information. 
If we take the example of iris data, if some people use different terms to record information for a specific column, such as naming the column `species` instead of `Species`, or using `iris setosa`, `set.` or `i.setosa` instead of `setosa`, the creation of a reproducible workflow  will be more difficult and errors may even be overlooked.  
Discrepancies often lead to errors, especially when the same terms could mean different things depending on who is entering the data. 
For example, indicating date as 02-03 will mean February the third in the USA, but March the second in Europe.


It is good practice to implement a `data dictionary`or a `taxonomy`of accepted terms and to document the convention used in a readme file.
Depending on the software you are using, you may be able to restrict the accepted values in specific columns.
If there is such a taxonomy or ontology available, using it may allow you (or others) to use the data in conjonction with other datasets.
For example, use the generic `male` and `female` terms for the sex of animal (without capitals, and without using abbreviation), as many ontologies use these terms.
In addition, you may want to use some extra tools to validate the spreadsheets before its integration in the analysis.


In addition, you should have clear rules about missing data points, using `NA`, `NULL` and/or empty cells is not trivial and may have different meanings (impossible data point, not recorded, or lost data point). 
Let's imagine, one would record the time spent before seeing the first pollinator landing on the iris flower measured previously, and that no pollinator was seen during the 5 minutes experiment.
If one reports `600` (the duration of the experiment in seconds), there will be no way to distinguish a case where no pollinator was seen, and one when one was seen at the end of the experiment (and you may forget that rule and treat 600 as a normal value).
If `NA` is reported, one may interpret this value as a non-existing data point (the experiment had not been performed).
An elegant solution is to have a second column stating whether a pollinator was seen during the experiment, where `TRUE`, `FALSE` and `NA` values are accepted.

Finally, you should also be aware of the default behaviour of your spreadsheet program, as it may be different for different programs, and for different versions of the same program.
For instance, the decimal is usually indicated with a comma in the French or German version of excel, but while a dot in enlish versions where the comma has no meaning
(`9,000`  will be translated into 9000 or 9 depending on the version you are using).
```

Reply:
this should resolve the consistency comment you had!

Reply:
Thank you so much. This does make it very clear.